### PR TITLE
rework init containers test to remove host file dependency

### PR DESCRIPTION
**Pull request by tzneal**
Since we can't rely on the test runner and hosts under test to be on the same machine, we write to the terminate log from each container and concatenate the results.

#### What type of PR is this?

/kind failing-test

#### What this PR does / why we need it:

Since we can't rely on the test runner and hosts under test to
be on the same machine, we write to the terminate log from each
container and concatenate the results.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [Test Grid]: https://testgrid.k8s.io/sig-node-cri-o#ci-crio-cgroupv1-node-e2e-conformance
```

### CODE REVIEW

File: test/e2e_node/init_containers_test.go [hunk 1]
```diff
@@ -54,13 +54,17 @@ type containerTestConfig struct {
...
+	// The busybox time command doesn't support sub-second display. uptime displays in hundredths of a second, so we
+	// include both and use time since boot for relative ordering
+	timeCmd := "`date +%s` `cat /proc/uptime | awk '{print $1}'`"
+	fmt.Fprintf(&cmd, "echo %s '%s Starting' >> /dev/termination-log; ", timeCmd, c.Name)
+	fmt.Fprintf(&cmd, "echo %s '%s Delaying %d' >> /dev/termination-log; ", timeCmd, c.Name, c.Delay)
 	if c.Delay != 0 {
 		fmt.Fprintf(&cmd, "sleep %d; ", c.Delay)
 	}
-	fmt.Fprintf(&cmd, "echo `date +%%s` '%s Exiting' >> /shared/output; ", c.Name)
+	fmt.Fprintf(&cmd, "echo %s '%s Exiting' >> /dev/termination-log; ", timeCmd, c.Name)
```
Comment (SergeyKanzhelev) [inline test/e2e_node/init_containers_test.go:67]: I'm not sure what implications might be for testing termination ordering. Should be covered I think. I think we can iterate - merging this to fix tests

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

File: test/e2e_node/init_containers_test.go (hunk 1 above)
Comment (tzneal) [reply test/e2e_node/init_containers_test.go:67]: Sounds good, was trying to come up with anyway to get data out of the containers without writing to the filesystem.  I thought about looking at container logs, but this seemed to be a bit more straight forward.

**Comment by harche** [comment]
Test log - https://prow.k8s.io/log?container=test&id=1633925899009134592&job=pull-kubernetes-node-crio-e2e

**Comment by gjkim42** [comment]
@tzneal 
How about using the container log instead of the termination log?
It is hard to use the termination log when testing the long-running container.

**Comment by tzneal** [comment]
It can probably be made to work. What causes the termination log to fail with the long running container?

**Comment by gjkim42** [comment]
> What causes the termination log to fail with the long running container?

It doesn't fail but we have to wait for its termination. It is hard to test the long-running container, as it takes a lot of time to see the termination-log. (something like sleep 600)

**Comment by gjkim42** [comment]
(When we just want to test the start time of the long-running container not the termination time)

**Comment by tzneal** [comment]
Ah, that makes sense.  Yeah I think its fine to switch to container logs if it makes it easier.  Shouldn't be difficult to refactor that way.

**Pull request merged**